File: JULY-2025/25-07-2025/p3487.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function maxSum($nums) {
        $map = [];
        $sum = 0;
        $maxN = -200;

        foreach ($nums as $value) {
            if (!in_array($value, $map)) {
                if ($value > 0) {
                    $map[] = $value;
                    $sum += $value;
                } else {
                    if ($value > $maxN)
                        $maxN = $value;
                }
            }
        }

        if ($sum == 0) return $maxN;
        else return $sum;
    }
}

File: JULY-2025/25-07-2025/p3487.kt
class Solution {
    fun maxSum(nums: IntArray): Int {
        val set = nums.toMutableSet()
        val negs = set.filter { it < 0 }
        
        if (negs.size == set.size) {
            return negs.max()
        } else {
            negs.forEach{set.remove(it)}
            return set.sum()
        }
    }
}